kratos/processes/structured_mesh_generator_process.cpp

```diff
@@ -232,6 +227,71 @@ namespace Kratos
 		}
 	}
 
+    void StructuredMeshGeneratorProcess::Check()
+    {
+        KRATOS_TRY
+
+        KRATOS_CHECK(CheckDomainGeometry());
+        KRATOS_CHECK(KratosComponents<Element>::Has(mElementName));
+
+        if ((mrGeometry.GetGeometryType() != GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4) &
+            (mrGeometry.GetGeometryType() != GeometryData::KratosGeometryType::Kratos_Hexahedra3D8))
+            KRATOS_ERROR << "An unsupported geometry was given. Only Quadrilateral2D4 and Hexahedra3D8 are supported and given geometry is : " << mrGeometry << std::endl;
+
+        KRATOS_CHECK_NOT_EQUAL(mNumberOfDivisions, 0);
+
+        KRATOS_CATCH("")
+    }
+
+    bool StructuredMeshGeneratorProcess::CheckDomainGeometry() {
+        if (mrGeometry.GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4) {
+            return CheckDomainGeometryConnectivityForQuadrilateral2D4();
+        }
+
+        else if (mrGeometry.GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Hexahedra3D8) {
+            return CheckDomainGeometryConnectivityForHexahedra3D8();
+        }
+
+        return true;
+    }
+
+    bool StructuredMeshGeneratorProcess::CheckDomainGeometryConnectivityForQuadrilateral2D4() {
+        return true; // TO DO: write check
```

Comment:
Would you please implement also this case?

Reply:
I have now done so